support/core.go

```diff
@@ -184,3 +184,33 @@ func GetNodeInternalIP(t Test, node corev1.Node) (IP string) {
 
 	return
 }
+
+func GetContainerName(t Test, container corev1.Container) string {
+	t.T().Helper()
+	t.Expect(container.Name).Should(gomega.Not(gomega.BeEmpty()))
+	return container.Name
+}
+
+func GetVolumeName(t Test, volume corev1.Volume) string {
+	t.T().Helper()
+	t.Expect(volume.Name).Should(gomega.Not(gomega.BeEmpty()))
+	return volume.Name
+}
+
+func GetServiceAccountName(t Test, serviceAccount corev1.ServiceAccount) string {
+	t.T().Helper()
+	t.Expect(serviceAccount.Name).Should(gomega.Not(gomega.BeEmpty()))
+	return serviceAccount.Name
+}
+
+func GetVolumeMountName(t Test, volumeMount corev1.VolumeMount) string {
+	t.T().Helper()
+	t.Expect(volumeMount.Name).Should(gomega.Not(gomega.BeEmpty()))
+	return volumeMount.Name
+}
+
+func GetEnvVarName(t Test, envVar corev1.EnvVar) string {
+	t.T().Helper()
+	t.Expect(envVar.Name).Should(gomega.Not(gomega.BeEmpty()))
+	return envVar.Name
+}
```

Comment:
Can be replaced by:
```
func ResourceName(meta metav1.Object) string {
	return meta.GetName()
}
```

Reply:
Just a note - for such helper functions I usually don't put `Get` prefix as the function doesn't query Kubernetes to get the object, just retrieving internal value.

Reply:
I don't quite understand how all those functions can be replaced by the `func ResourceName(meta metav1.Object)`. They are of different type where we would be getting the name of the top-level resource such as a pod, as opposed to the embedded resources.

Reply:
Right, I didn't check that before. Unfortunately it seems they can't be replaced.

Reply:
We can attempt to replace all functions for:
```
func ResourceName(t Test, obj any) string {
	t.T().Helper()
	t.Expect(reflect.ValueOf(obj).Kind()).Should(gomega.Equal(reflect.Struct), "input must be a struct")
	nameField := reflect.ValueOf(obj).FieldByName("Name")
	return nameField.String()
}
```